measure/src/main/scala/org/apache/griffin/measure/datasource/connector/batch/JDBCBasedDataConnector.scala

```diff
@@ -0,0 +1,127 @@
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import org.apache.griffin.measure.Loggable
+import org.apache.griffin.measure.configuration.dqdefinition.DataConnectorParam
+import org.apache.griffin.measure.context.TimeRange
+import org.apache.griffin.measure.datasource.TimestampStorage
+import org.apache.griffin.measure.utils.ParamUtil._
+
+/**
+ * A batch data connector for JDBC based source which allows support for various
+ * JDBC based data sources like Oracle. Postgres etc.
+ *
+ * Supported Configurations:
+ *  - database : [[String]] specifying the database name.
+ *  - tablename : [[String]] specifying the table name to be read
+ *  - url : [[String]] specifying the URL to connect to database
+ *  - user : [[String]] specifying the user for connection to database
+ *  - password: [[String]] specifying the password for connection to database
+ *  - driver : [[String]] specifying the driver for JDBC connection to database
+ *  - where : [[String]] specifying the condition for reading data from table
+ *
+ * Some defaults assumed by this connector (if not set) are as follows:
+ *  - `database` is default,
+ *  - `driver` is com.mysql.jdbc.Driver,
+ *  - `where` is None
+ */
+case class JDBCBasedDataConnector(
+    @transient sparkSession: SparkSession,
+    dcParam: DataConnectorParam,
+    timestampStorage: TimestampStorage)
+    extends BatchDataConnector {
+
+  import JDBCBasedDataConnector._
+  val config: Map[String, Any] = dcParam.getConfig
+  val database: String = config.getString(Database, DefaultDatabase)
+  val tableName: String = config.getString(TableName, EmptyString)
+  val fullTableName: String = s"$database.$tableName"
+  val whereString: String = config.getString(Where, EmptyString)
+  val url: String = config.getString(Url, EmptyString)
+  val user: String = config.getString(User, EmptyString)
+  val password: String = config.getString(Password, EmptyString)
+  val driver: String = config.getString(Driver, DefaultDriver)
+
+  require(url.nonEmpty, "JDBC connection: connection url is mandatory")
+  require(user.nonEmpty, "JDBC connection: user name is mandatory")
+  require(password.nonEmpty, "JDBC connection: password is mandatory")
+  require(tableName.nonEmpty, "JDBC connection: table is mandatory")
+  assert(isJDBCDriverLoaded(driver), s"JDBC driver ${driver} not present in classpath")
+
+  override def data(ms: Long): (Option[DataFrame], TimeRange) = {
+    val dfOpt = try {
+      val dtSql = createSqlStmt()
+      val prop = new java.util.Properties
+      prop.setProperty("user", user)
+      prop.setProperty("password", password)
+      prop.setProperty("driver", driver)
+      val df: DataFrame = sparkSession.read.jdbc(url, s"($dtSql) as t", prop)
+      val dfOpt = Some(df)
+      val preDfOpt = preProcess(dfOpt, ms)
+      preDfOpt
+    } catch {
+      case e: Throwable =>
+        error(s"load mysql table $fullTableName fails: ${e.getMessage}", e)
```

Comment:
`mysql` keyword should not be included in error messages.

Reply:
Hi @wankunde,
Made the changes as per review comments in 2ee6869500b813a2a95f56e5ba3b242cea711ffd, please review it.